src/chess/piece/Bishop.java

```diff
@@ -15,8 +14,8 @@ public Bishop(Color color) {
   }
 
   @Override
-  public boolean isValidMove(Position newPosition) {
-    if (!super.isValidMove(newPosition)) {
+  public boolean isValidMove(int row, int col, boolean isEnemyPiece) {
```

Comment:
I thought the name `isEnemyPiece` is a bit unclear
Instead of that, just for example, I might name like `isEnemyExisted`